### ISSUE TITLE
CA2014 misfires

### ISSUE THREAD

**Issue by soerenwolfers** [dotnet/roslyn-analyzers#6283]
### Description

Stackallocating in the "header" (i.e., not the body) of a `foreach` loop gives a "CA2014: Do not use stackalloc in loops" warning.

### Reproduction Steps

```
public static void Foo()
{
    foreach (var x in stackalloc double[] { 0 }) 
    {
       Console.WriteLine(x);
    }
} 
```

### Expected behavior

No warnings, because the allocation only happens once per function call, not once per iteration of the loop body.

### Actual behavior

"CA2014: Do not use stackalloc in loops" warning.

### Regression?

Warning introduced in https://github.com/dotnet/roslyn-analyzers/pull/3432

### Known Workarounds

```
#pragma warning disable CA2014
...
#pragma warning restore CA2014
```

**Comment by Youssef1313** [comment]
@soerenwolfers What version of Microsoft.CodeAnalysis.NetAnalyzers are you using? It seems this was fixed in https://github.com/dotnet/roslyn-analyzers/pull/5700 already

**Comment by Youssef1313** [comment]
A possible non-fixed bug is:

```csharp
public class C
{  
    public static unsafe void Foo()
    {
        foreach (var (x, y) in stackalloc (double, double)[] { (0, 0) })
        {
       
        }
    } 
}

```

**Comment by mavasani** [comment]
Verified that this does not reproduce with the latest SDK analyzers.
